jax/interpreters/xla.py

```diff
@@ -372,6 +376,18 @@ def _check_nans(name, xla_shape, buf):
     if np.any(np.isnan(buf.to_py())):
       raise FloatingPointError(f"invalid value (nan) encountered in {name}")
 
+def check_infs(prim, bufs):
```

Comment:
I am wondering if instead of duplicating the code for NaN, we can extend it to check for any combination of NaN and Inf based on the flags?

Reply:
I agree on the duplication, but wasn't confident doing major surgery on a first PR. I'll try out a little refactor and see what it looks like though.